From ece4d947760b0ebe1ada4ba7ff5118c407ef272f Mon Sep 17 00:00:00 2001
From: Alexandru Costache <alexandru@balena.io>
Date: Thu, 23 Feb 2023 17:32:31 +0100
Subject: [PATCH] linux-compulab: Don't switch overlayfs as module from
 defconfig

The defconfig seems to sporadically override the
setting from meta-balena, let's ensure this module
is built-in to avoid kernel panics that sometimes
don't print enough information to diagnose the actual
root cause.

Signed-off-by: Alexandru Costache <alexandru@balena.io>
---
 arch/arm64/configs/iot-gate-imx8plus_defconfig | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/configs/iot-gate-imx8plus_defconfig b/arch/arm64/configs/iot-gate-imx8plus_defconfig
index 575741d2e1e8..30046c4cf355 100644
--- a/arch/arm64/configs/iot-gate-imx8plus_defconfig
+++ b/arch/arm64/configs/iot-gate-imx8plus_defconfig
@@ -1273,6 +1273,9 @@ CONFIG_EXT2_FS_SECURITY=y
 CONFIG_EXT3_FS=y
 CONFIG_EXT3_FS_POSIX_ACL=y
 CONFIG_EXT3_FS_SECURITY=y
+CONFIG_EXT4_FS=y \
+CONFIG_EXT4_FS_POSIX_ACL=y \
+CONFIG_EXT4_FS_SECURITY=y \
 CONFIG_BTRFS_FS=m
 CONFIG_BTRFS_FS_POSIX_ACL=y
 CONFIG_FANOTIFY=y
@@ -1281,7 +1284,7 @@ CONFIG_QUOTA=y
 CONFIG_AUTOFS4_FS=y
 CONFIG_FUSE_FS=m
 CONFIG_CUSE=m
-CONFIG_OVERLAY_FS=m
+CONFIG_OVERLAY_FS=y
 CONFIG_VFAT_FS=y
 CONFIG_TMPFS_POSIX_ACL=y
 CONFIG_HUGETLBFS=y
-- 
2.37.2

